### PR TITLE
Pushes waiting for enclaves start-up to inside host. More graceful handling of connection errors.

### PR DESCRIPTION
**Pull request by joeldudley**
We were panicking too much on connection errors before. There were lots of spurious errors at the end of the simulation, when the listeners were being closed and failed connections are to be accepted.

### CODE REVIEW

**Comment by otherview** [comment]
What if these listeners and dialers had a graceful shutdown ?

**Review by tudor-malene — APPROVED** [review]
lgtm

File: go/obscuronode/host/p2p/p2p.go [hunk 1]
```diff
@@ -101,14 +98,19 @@ func (p *p2pImpl) BroadcastRollup(bytes []byte) {
 }
 
 // Listens for connections and handles them in a separate goroutine.
-func (p *p2pImpl) handleConnections(txP2PCh chan nodecommon.EncryptedTx, rollupsP2PCh chan obscurocommon.EncodedRollup, listener net.Listener) {
+func (p *p2pImpl) handleConnections(interrupt chan bool, txP2PCh chan nodecommon.EncryptedTx, rollupsP2PCh chan obscurocommon.EncodedRollup, listener net.Listener) {
```
Comment (otherview) [inline go/obscuronode/host/p2p/p2p.go:101]: Do we need to pass the interrupt channel if it's on the struct ?

File: go/obscuronode/host/p2p/p2p.go [hunk 2]
```diff
@@ -60,35 +61,34 @@ func NewP2P(ourAddress string, allAddresses []string) P2P {
...
 
 func (p *p2pImpl) StopListening() {
-	if p.txListener != nil {
-		if err := p.txListener.Close(); err != nil {
-			log.Log(fmt.Sprintf("failed to close transaction P2P listener cleanly: %v", err))
-		}
-		p.txListener = nil
-	}
+	i := int32(1)
+	p.listenerInterrupt = &i
```
Comment (tudor-malene) [inline go/obscuronode/host/p2p/p2p.go:86]: atomic.StoreInt32(p.listenerInterrupt, 1)

**Review by tudor-malene — CHANGES_REQUESTED** [review]
you're not setting the interrupt correctly

File: go/obscuronode/host/p2p/p2p.go [hunk 3]
```diff
@@ -60,35 +61,33 @@ func NewP2P(ourAddress string, allAddresses []string) P2P {
...
 		panic(err)
 	}
-	p.txListener = txListener
-	go p.handleConnections(txP2PCh, rollupsP2PCh, txListener)
+
+	i := int32(0)
+	p.listenerInterrupt = &i
+	p.listener = listener
+
+	go p.handleConnections(txP2PCh, rollupsP2PCh, listener)
```
Comment (tudor-malene) [inline go/obscuronode/host/p2p/p2p.go:81]: there is no need to pass the listener here, since it's in the p2pimpl anyway

**Review by tudor-malene — APPROVED** [review]
Lgtm.
One minor thing, if you want to fix it

**Pull request merged**